src/workerd/jsg/commonjs.c++

```diff
@@ -0,0 +1,163 @@
+#include "commonjs.h"
+
+#include "modules.h"
+
+namespace workerd::jsg {
+
+v8::Local<v8::Value> CommonJsModuleContext::require(jsg::Lock& js, kj::String specifier) {
+  auto modulesForResolveCallback = getModulesForResolveCallback(js.v8Isolate);
+  KJ_REQUIRE(modulesForResolveCallback != nullptr, "didn't expect resolveCallback() now");
+
+  if (isNodeJsCompatEnabled(js)) {
+    KJ_IF_SOME(nodeSpec, checkNodeSpecifier(specifier)) {
+      specifier = kj::mv(nodeSpec);
+    }
+  }
+
+  kj::Path targetPath = ([&] {
```

Comment:
Can we move this function outside? this way it won't get inlined, and I assume this will run a lot, right?

Reply:
For this PR I'm intentionally just moving things around, not changing the current impl. I'm planning some other changes here that will come soon in a different PR